src/main/java/executor/service/service/ParallelFlowExecutorService.java

```diff
@@ -0,0 +1,76 @@
+package executor.service.service;
+
+import executor.service.config.beans.ThreadPoolFactory;
+import executor.service.handler.ProxyHandler;
+import executor.service.handler.ScenarioHandler;
+import executor.service.model.ProxyConfigHolder;
+import executor.service.model.Scenario;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+
+public class ParallelFlowExecutorService {
+
+    private static final int CAPACITY = 50;
+    private static final Logger log = LoggerFactory.getLogger(ParallelFlowExecutorService.class);
+    private ExecutionService executionService;
+    private final ScenarioSourceListener scenarioListener;
+    private final ProxySourcesClient proxyClient;
+    private final ExecutorService threadPoolExecutor;
+    private BlockingQueue<Scenario> scenarioQueue;
+    private BlockingQueue<ProxyConfigHolder> proxyQueue;
+
+
+    public ParallelFlowExecutorService(
+            ExecutionService executionService,
+            ScenarioSourceListener scenarioListener,
+            ProxySourcesClient proxyClient) {
+        this.executionService = executionService;
+        this.scenarioListener = scenarioListener;
+        this.proxyClient = proxyClient;
+        this.threadPoolExecutor = ThreadPoolFactory.getThreadPoolExecutor();
+        this.scenarioQueue = new LinkedBlockingQueue<>(CAPACITY);
+        this.proxyQueue = new LinkedBlockingQueue<>(CAPACITY);
+    }
+
+    public void run() {
+        threadPoolExecutor.submit(() -> scenarioListener.execute(getScenarioHandler()));
```

Comment:
Why are you calling the "**submit()**" method here, are you planning to somehow process the result of the workflow, then where is this processing? You can use **execute()** - thus **not confusing** the developers.

Reply:
ok